### DISCUSSION TITLE
Make use of the new polaris algorithm ?

### DISCUSSION THREAD

**Discussion by LifeIsStrange** [servo/servo#9970] [General]
For Faster load 
https://www.csail.mit.edu/node/2720

**Comment by frewsxcv** [comment]
Media coverage about this: http://www.zdnet.com/article/mits-answer-to-cutting-webpage-load-times-its-the-polaris-compression-trumping-browser-framework/

**Comment by larsbergstrom** [comment]
Link to the paper here: http://web.mit.edu/ravinet/www/polaris_nsdi16.pdf

Certainly, I'd be open to seeing a research group evaluate the results within Servo! Some interesting questions to me are:
- Now that HTTP/2 has rolled out more significantly since the analysis was done, how does Polaris compare to HTTP/2?
- What is the impact on power consumption? Browser network load is optimized both for first page load and (on mobile) to avoid thrashing the modem on mobile phones. How does this framework affect it?
- How much will this help normal user browsing patterns? It requires analysis of a full page load and the dependency graph in order to provide speedups, but as section 5.3/Figure 10 show, Polaris has reduced effectiveness when the user browser cache is warm.
- One might like to share this dependency data between users, but privacy. How does any information gleamed from this interact with privacy concerns, the move to https-everywhere, private browsing mode, etc.?

Anyway, sorry if I come across as "reviewer #2" here - those are just some of the questions I'd like to see somebody look into if they wanted to experiment with this approach.

**Comment by gsnedders** [comment]
> One might like to share this dependency data between users, but privacy. How does any information gleamed from this interact with privacy concerns, the move to https-everywhere, private browsing mode, etc.?

I wonder if basing this upon HTTP caching headers makes sense, and using the notion of shared caches? Obviously one would need to consider cache headers for the entire graph, which presumably increases the probability of one resource not being cachable.
